espn_api/football/fantasy.py

```diff
@@ -0,0 +1,172 @@
+from espn_api.football import League
+from operator import attrgetter
+
+# Add award to dict of teams
+def award(team_name, award_string):
+	if awards.get(team_name) != None:
+		awards[team_name].append(award_string)
+	else: 
+		awards[team_name] = [award_string]
+
+# Print all awards for each team
+def printAwards(awards):
+	for team in awards:
+		print(team)
+		for award in awards[team]:
+			print(award)
+		print()
+
+# Compute highest value for given position(s)
+def computeHigh(pos, players, award_string):
+
+	# Compile list of players at position(s) pos
+	filtered_players = [x for x in players if x.lineupSlot in pos]
+	filtered_dict = {}
+
+	# Make a dictionary of team_name -> sum of players at that position
+	for team in league.teams:
+		filtered_dict[team.team_name] = sum(player.points for player in filtered_players if player.onTeamId == team.team_id)
+	
+	# Compute team with highest value
+	top_team = max(filtered_dict, key=filtered_dict.get)
+	player_myst = whoWasThat(top_team, filtered_dict[top_team], filtered_players)
+	award_string = award_string + player_myst + str(filtered_dict[top_team]) + ')'
+	award(top_team, award_string)
+
+# Attempt to match what player did the thing 
+def whoWasThat(team_name, score, players):
+	team = getTeam(team_name)
+	for player in players:
+		# print(player.points)
+		if player.points == score and player.onTeamId == team.team_id:
+			return player.name.rsplit(' ', 1)[1] + ', '
+	return ''
+
+# Get team name for a team_id
+def getTeamName(id):
+	return next((y.team_name for y in league.teams if y.team_id == id), None)
+
+def getTeam(name):
+	return next((y for y in league.teams if y.team_name == name), None)
+
+LEAGUE_ID = 306883
+YEAR = 2024
+league = League(LEAGUE_ID, YEAR)
+awards = {}
+players = []
+
+week = 5
+box_scores = league.box_scores(week=week)
+
+week_high = 0
+week_high_team = ''
+week_low = 200
+week_low_team = ''
+week_high_diff = 0
+week_low_diff = 200
+diff_high_team = ''
+diff_low_team = ''
+winners = {}
+losers = {}
+
+# Iterating over matchups
+for matchup in box_scores:
```

Comment:
You could clean this loop up a bit by iterating over the box scores first and hashing team name to score

```
for matchup in box_scores:
    scores[matchup.away_team.team_name] = matchup.away_score
    scores[matchup.home_team.team_name] = matchup.home_score
```

Then for this loop you could iterate over that dictionary and it would save you having to do everything twice for home/away.